apps/web/components/utils.tsx

```diff
@@ -241,16 +243,46 @@ export async function createTrack(data: {
   }
 }
 
-export async function updateTrack(trackId: string, data: any) {
+export async function updateTrack(trackId: string, data: {
+  id: string;
+  title: string;
+  description: string;
+  image: string;
+  selectedCategory?: string[];
+  problems?: { problem: Prisma.ProblemCreateManyInput; sortingOrder: number }[];
+  hidden: boolean;
+}) {
   try {
     const track = await db.track.update({
       where: {
         id: trackId,
       },
-      data,
+      data:{
+        id: data.id,
+        title: data.title,
+        description: data.description,
+        image: data.image,
+        hidden: data.hidden,
```

Comment:
Don't we need to add selectedCategory : data.selectedCategory

Reply:
No, we don't as I'm doing it below in the same function